apps/aggregator/src/snapshots.service.ts

```diff
@@ -0,0 +1,187 @@
+import { Injectable } from '@nestjs/common';
+import { AlertsService, ApiConfigService, LockedEgldProvider } from '@libs/common';
+import BigNumber from 'bignumber.js';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { BaseProvider } from '../../../providers/base.provider';
+import { loadProvider } from '../../../common/provider.loader';
+import { Lock, OriginLogger } from '@multiversx/sdk-nestjs-common';
+import { ElasticIndexerService } from './elastic/elastic.indexer.service';
+import { JsonExporterService } from './jsonExporter/json.exporter.service';
+import { ApiService } from '@multiversx/sdk-nestjs-http';
+
+@Injectable()
+export class SnapshotsService {
+  private readonly logger = new OriginLogger(SnapshotsService.name);
+  private readonly API_SLEEP_TIME = 100;
+
+  constructor(
+    private readonly baseProvider: BaseProvider,
+    private readonly apiService: ApiService,
+    private readonly apiConfigService: ApiConfigService,
+    private readonly alertsService: AlertsService,
+    private readonly elasticIndexer: ElasticIndexerService,
+    private readonly jsonExporter: JsonExporterService,
+  ) { }
+
+  @Cron(CronExpression.EVERY_DAY_AT_10AM)
```

Comment:
can we have a random time so we don`t have users that tries to play the snapshot process in their favor

Reply:
currently it's just a placeholder and will be changed when deploying the service